leading forward slash of inline comment removed

<!--
This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes
3. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**Tell us about your environment**
* Window 7 
* **ESLint Version: v3.6.1**
* **Node Version: v4.4.7**
* **npm Version: 3.10.3**

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<!-- Paste your configuration below: -->
```
{
  "env": {
    "browser": true,
    "amd": true
  },
  "globals": {},
  "rules": {
    "semi": [
      "error",
      "always"
    ],
    "no-bitwise": 0,
    "no-cond-assign": 0,
    "camelcase": 2,
    "curly": 2,
    "no-debugger": 0,
    "eqeqeq": 2,
    "no-eq-null": 2,
    "no-eval": 0,
    "no-unused-expressions": 2,
    "guard-for-in": 0,
    "block-scoped-var": 2,
    "strict": 0,
    "comma-dangle": [
      "error",
      "never"
    ],
    "lines-around-directive": [
      "error",
      {
        "before": "never",
        "after": "always"
      }
    ],
    "object-curly-newline": [
      "error",
      "always"
    ],
    "wrap-iife": [
      2,
      "any"
    ],
    "keyword-spacing": [
      "error",
      {
        "before": true
      }
    ],
    "key-spacing": [
      "error",
      {
        "beforeColon": false
      }
    ],
    "no-iterator": 0,
    "no-use-before-define": 0,
    "comma-style": [
      2,
      "last"
    ],
    "space-before-blocks": [
      2,
      "always"
    ],
    "space-before-function-paren": [
      2,
      "always"
    ],
    "no-trailing-spaces": [
      2,
      {
        "skipBlankLines": true
      }
    ],
    "space-infix-ops": [
      2,
      {
        "int32Hint": false
      }
    ],
    "no-mixed-spaces-and-tabs": 2,
    "no-loop-func": 2,
    "no-multi-str": 0,
    "new-cap": 2,
    "no-caller": 2,
    "no-empty": 0,
    "no-new": 2,
    "no-plusplus": 0,
    "no-proto": 0,
    "no-script-url": 2,
    "no-shadow": 2,
    "dot-notation": 2,
    "no-new-func": 0,
    "no-new-wrappers": 0,
    "no-undef": 2,
    "no-unused-vars": 2,
    "no-invalid-this": 0,
    "no-with": 0,
    "max-len": [
      2,
      {
        "code": 120,
        "ignoreComments": true
      }
    ],
    "indent": [
      "error",
      "tab"
    ],
    "quotes": [
      "error",
      "double",
      {
        "avoidEscape": true
      }
    ],
    "quote-props": [
      2,
      "as-needed",
      {
        "unnecessary": true
      }
    ],
    "complexity": [
      2,
      10
    ]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

ran `eslint filename.js --fix`

<!-- Paste the source code below: -->
```js
modules.exports = {
	foo: function (arg1, arg2, arg3) {},
	bar: /*static*/ function (arg1, arg2, arg3) {},
	baz /*static*/ : function (arg1, arg2, arg3) {}
};
```

**What did you expect to happen?**

The inline comment after **baz** to remain intact.

```js
modules.exports = {
	foo: function (arg1, arg2, arg3) {},
	bar: /*static*/ function (arg1, arg2, arg3) {},
	baz /*static*/ : function (arg1, arg2, arg3) {}
};
```

**What actually happened? Please include the actual, raw output from ESLint.**

```sh
  61:5  error  Parsing error: Unexpected token *

✖ 1 problem (1 error, 0 warnings)
```

The forward slash after **baz** has been removed
```js

modules.exports = {
	foo: function (arg1, arg2, arg3) {},
	bar: /*static*/ function (arg1, arg2, arg3) {},
	baz*static*/ : function (arg1, arg2, arg3) {}
};
```

--- a/lib/rules/key-spacing.js
+++ b/lib/rules/key-spacing.js
@@ -417,8 +417,8 @@ module.exports = {
         function report(property, side, whitespace, expected, mode) {
             const diff = whitespace.length - expected,
                 nextColon = getNextColon(property.key),
-                tokenBeforeColon = sourceCode.getTokenBefore(nextColon),
-                tokenAfterColon = sourceCode.getTokenAfter(nextColon),
+                tokenBeforeColon = sourceCode.getTokenOrCommentBefore(nextColon),
+                tokenAfterColon = sourceCode.getTokenOrCommentAfter(nextColon),
                 isKeySide = side === "key",
                 locStart = isKeySide ? tokenBeforeColon.loc.start : tokenAfterColon.loc.start,
                 isExtra = diff > 0,

--- a/tests/lib/rules/key-spacing.js
+++ b/tests/lib/rules/key-spacing.js
@@ -1739,5 +1739,14 @@ ruleTester.run("key-spacing", rule, {
         output: "({ foo:/* comment */bar })",
         options: [{ afterColon: false }],
         errors: [{ message: "Extra space before value for key 'foo'.", line: 1, column: 9, type: "Identifier" }]
+    },
+    {
+        code: "({ foo/*comment*/:/*comment*/bar })",
+        output: "({ foo/*comment*/ : /*comment*/bar })",
+        options: [{ beforeColon: true, afterColon: true }],
+        errors: [
+            { message: "Missing space after key 'foo'.", line: 1, column: 7, type: "Identifier" },
+            { message: "Missing space before value for key 'foo'.", line: 1, column: 19, type: "Identifier"}
+        ]
     }]
 });

--- a/tests/lib/rules/key-spacing.js
+++ b/tests/lib/rules/key-spacing.js
@@ -1728,5 +1728,16 @@ ruleTester.run("key-spacing", rule, {
             { message: "Extra space before value for key 'key2'.", line: 4, column: 14, type: "Literal" },
             { message: "Extra space before value for key 'key3'.", line: 5, column: 14, type: "Literal" }
         ]
+    }, {
+
+        // https://github.com/eslint/eslint/issues/7603
+        code: "({ foo/* comment */ : bar })",
+        output: "({ foo/* comment */: bar })",
+        errors: [{ message: "Extra space after key 'foo'.", line: 1, column: 7, type: "Identifier" }]
+    }, {
+        code: "({ foo: /* comment */bar })",
+        output: "({ foo:/* comment */bar })",
+        options: [{ afterColon: false }],
+        errors: [{ message: "Extra space before value for key 'foo'.", line: 1, column: 9, type: "Identifier" }]
     }]
 });

